platform/logger_helpers.cpp

```diff
@@ -0,0 +1,161 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// No logging supported in case of NDEBUG or ID based
+#if !defined(NDEBUG) && !defined(MBED_ID_BASED_TRACING)
+
+#include <stdint.h>
+#include "platform/mbed_assert.h"
+#include "platform/PlatformMutex.h"
+#include "platform/SingletonPtr.h"
+#include "platform/mbed_critical.h"
+#include "platform/logger_internal.h"
+#if defined(MBED_CONF_MBED_TRACE_FEA_IPV6)
+#include "mbed-client-libservice/ip6string.h"
+#include "mbed-client-libservice/common_functions.h"
+#endif
+
+#define LOG_SINGLE_HELPER_STR_SIZE_     LOG_SINGLE_STR_SIZE_
+
+static SingletonPtr<PlatformMutex> log_helper_lock;
+static uint32_t log_mutex_count = 0;
+static char log_helper_data[LOG_SINGLE_HELPER_STR_SIZE_];
+static char send_null[] = "";
+static char send_null_str[] = "<null>";
```

Comment:
These are wasting RAM. Even if they were `const`, still not a brilliant idea to try to pre-empt the compiler by forcing the static allocation. Just use string literals and let the compiler do its job.